brokers/brokers.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package brokers
+
+import (
+	context "context"
+
+	"github.com/mainflux/mainflux"
+	"github.com/nats-io/nats.go"
+)
+
+// MessagePublisher specifies a message publishing API.
+type MessagePublisher interface {
```

Comment:
I would consider just Publisher and Subscriber, maybe we don't need Message prefix (its a broker package Publisher.Publish() is obvious). Like io.Reader and io.Writer simple.